src/Blocks/BlockRail.h

```diff
@@ -34,16 +34,27 @@ class cBlockRailHandler :
 		BLOCKTYPE & a_BlockType, NIBBLETYPE & a_BlockMeta
 	) override
 	{
-		a_BlockType = m_BlockType;
+		/** A rail cannot be attached to the side or bottom of any block,
+		but attempting to make such an attachment may cause the rail
+		to attach to the top of a block under the destination space */
+
 		Vector3i Pos{ a_BlockX, a_BlockY, a_BlockZ };
-		a_BlockMeta = FindMeta(a_ChunkInterface, Pos);
-		return a_Player.GetWorld()->DoWithChunkAt(Pos,
-			[this, Pos, &a_ChunkInterface](cChunk & a_Chunk)
-			{
-				auto RelPos = cChunkDef::AbsoluteToRelative(Pos);
-				return CanBeAt(a_ChunkInterface, RelPos.x, RelPos.y, RelPos.z, a_Chunk);
-			}
-		);
+		bool CheckResult = a_Player.GetWorld()->DoWithChunkAt(Pos, [&](cChunk & a_Chunk)
+		{
+			auto RelPos = cChunkDef::AbsoluteToRelative(Pos);
+			return CanBeAt(a_ChunkInterface, RelPos.x, RelPos.y, RelPos.z, a_Chunk, 0);
```

Comment:
This call is not needed, sorry I miss this modification.
Pass 0 to meta because meta is not needed in **CanBeAt**.

I will modify **GetPlacementBlockTypeMeta**